src/System.Text.RegularExpressions/tests/RegexCompilationHelper.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace System.Text.RegularExpressions.Tests
+{
+    public class RegexCompilationHelper
+    {
+        public static IEnumerable<object[]> TransformRegexOptions(string testDataMethodName, int regexOptionsArrayIndex)
+        {
+            // On Uap or NetNative the compiled feature isn't currently enabled,
+            // therefore we don't need the additional test data.
+            if (PlatformDetection.IsUap)
```

Comment:
AGain might be clearer to have a property like SupportsCompiledRegex or similar?

Reply:
I don't think that we need that property anywhere else besides this assembly. I wouldn't place it in PlatformDetection and instead prefer the current condition as it's the only place where we need to check that...